engine/access/rpc/engine_builder.go

```diff
@@ -88,10 +103,13 @@ func (builder *RPCEngineBuilder) Build() (*Engine, error) {
 	}
 	handler := builder.handler
 	if handler == nil {
+		if builder.finalizedHeaderCache == nil {
```

Comment:
given this is a required field, I think it should be a required argument in the constructor:
https://github.com/onflow/flow-go/blob/9b9625148522ddb1c6cd7fbfac14756531358724/engine/access/rpc/engine.go#L71-L91

Reply:
ah, I see. this rpc engine builder logic is super messy, but fixing that is definitely out of scope for this PR.

Reply:
re: the NodeID, the simplest thing to do is pass `me module.Local` into the rpc engine, which could then be passed to the handler. Then the node's ID is just `h.me.NodeID()`